content/courses/advanced-ada/chapters/access_types.rst

```diff
@@ -22,14 +24,488 @@ Access to subprograms
+          2 => Tab_2'Access,
+          3 => Tab_3'Access);
+
+    end Data_Processing;
+
+Here, instead of allocating the constant arrays dynamically via :ada:`new`, we
+declare three aliased arrays (:ada:`Tab_1`, :ada:`Tab_2` and :ada:`Tab_3`) and
+get a reference to them in the declaration of :ada:`Table`.
+
+
+Aliased access objects
+^^^^^^^^^^^^^^^^^^^^^^
+
+It's interesting to mention that access objects can be aliased themselves.
+Consider this example where we declare the :ada:`Integer_Access_Access` type
+to refer to an access object:
+
+.. code:: ada run_button project=Courses.Advanced_Ada.Access_Types.Aliased_Access
+
+    with Ada.Text_IO; use Ada.Text_IO;
+
+    procedure Show_Aliased_Access_Obj is
+
+       type Integer_Access        is access all Integer;
+       type Integer_Access_Access is access all Integer_Access;
+
+       I_Var : aliased Integer;
+       A     : aliased Integer_Access;
+       B     : Integer_Access_Access;
+    begin
+       A := I_Var'Access;
+       B := A'Access;
+
+       B.all.all := 22;
+       Put_Line ("A: " & Integer'Image (A.all));
+       Put_Line ("B: " & Integer'Image (B.all.all));
+    end Show_Aliased_Access_Obj;
+
+After the assignments in this example, :ada:`B` refers to :ada:`A`, which in
+turn refers to :ada:`I_Var`. Note that this code only compiles because we
+declare :ada:`A` as an aliased (access) object.
+
+
+Aliased components
+~~~~~~~~~~~~~~~~~~
+
+Components of an array or a record can be aliased. This allows us to get access
+to those components:
+
+.. code:: ada run_button project=Courses.Advanced_Ada.Access_Types.Aliased_Components
+
+    with Ada.Text_IO; use Ada.Text_IO;
+
+    procedure Show_Aliased_Components is
+
+       type Integer_Access is access all Integer;
+
+       type Rec is record
+          I_Var_1 :         Integer;
+          I_Var_2 : aliased Integer;
+       end record;
+
+       type Integer_Array is array (Positive range <>) of
+         aliased Integer;
+
+       R   : Rec := (22, 24);
+       Arr : Integer_Array (1 .. 3) := (others => 42);
+       A   : Integer_Access;
+    begin
+       --  A := R.I_Var_1'Access;
+       --                 ^ ERROR: cannot access
+       --                          non-aliased component
+
+       A := R.I_Var_2'Access;
+       Put_Line ("A: " & Integer'Image (A.all));
+
+       A := Arr (2)'Access;
+       Put_Line ("A: " & Integer'Image (A.all));
+    end Show_Aliased_Components;
+
+In this example, we get access to the :ada:`I_Var_2` component of record
+:ada:`R`. (Note that trying to access the :ada:`I_Var_1` component would gives us
+a compilation error, as this component is not aliased.) Similarly, we get
+access to the second component of array :ada:`Arr`.
+
+Declaring components with the :ada:`aliased` keyword allows us to specify that
+those are accessible via other paths besides the component name. Therefore, the
+compiler won't store them in registers. This can be essential when doing with
```

Comment:
In "when doing with" the word "with" should not be present.

Reply:
I assume I was intending to write "dealing with low-level programming." Anyway, I've just removed the "with" word.